### PR TITLE
Implemented demo endpoint and component

### PR DESCRIPTION
**Pull request by roll**
- connects #248 
---
gt

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
@danfowler 
please take a look

File: goodtablesio/settings.py [hunk 1]
```diff
@@ -24,6 +24,9 @@
 GLOB_EXCLUDED_FORMATS = ['json', 'sql']
 GOOGLE_ANALYTICS_CODE = os.environ.get('GOOGLE_ANALYTICS_CODE')
 MAX_TABLES_PER_SOURCE = 100
+DEMO_API_URL = '%s/api' % BASE_URL
+DEMO_API_TOKEN = 'D0123458B8E36326C60253FE4A7FF6662CAB0C48'
```
Comment (amercader) [inline goodtablesio/settings.py]: We should not hardcode this in the source code, so it's easy to change. Rather read it from an env var in `settings`

File: migrations/versions/20170713170632_add_demo_user.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    op.execute("""
+        INSERT INTO users (id, name, display_name, created, admin)
+        VALUES ('77af91ac-4ba2-4ecd-88eb-8ce650b62cce', 'demo', 'demo', now(), false)
+    """)
+    op.execute("""
+        INSERT INTO api_tokens (id, user_id, token, description, created)
+        VALUES (
+            'c8e1878a-f6da-404a-ac0c-e86e5607d497',
+            '77af91ac-4ba2-4ecd-88eb-8ce650b62cce',
+            'D0123458B8E36326C60253FE4A7FF6662CAB0C48',
```
Comment (amercader) [inline migrations/versions/20170713170632_add_demo_user.py]: Same

File: migrations/versions/20170713170632_add_demo_user.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            '77af91ac-4ba2-4ecd-88eb-8ce650b62cce',
+            'default'
+        )
+    """)
+
+
+def downgrade():
+    op.execute("DELETE FROM jobs WHERE source_id = '9b6b6391-5404-4e7f-bdb8-271c2cb42fbb'")
+    op.execute("DELETE FROM sources WHERE id = '9b6b6391-5404-4e7f-bdb8-271c2cb42fbb'")
+    op.execute("DELETE FROM users WHERE id = '77af91ac-4ba2-4ecd-88eb-8ce650b62cce'")
```
Comment (amercader) [inline migrations/versions/20170713170632_add_demo_user.py]: `DELETE FROM api_tokens`?

Comment (roll) [inline migrations/versions/20170713170632_add_demo_user.py]: `api_tokens` and `source_users` are on `DELETE CASCADE`

**Comment by roll** [comment]
@amercader 
So about token. What we do here:
- we create system user called `demo`
- we don't have any access to this user tokens via UI
- it means only proper way to add `demo` user token is using migrations
- but this demo token is PUBLIC by definition (so it's OK to have it in codebase)
- and I suppose there is no scenario when we need to change this token. 
This token is like marker that `demo` user uses API. Then we could add limits etc. We could share it with partners for testing etc. But I think the nature of this token is to be hard-coded because it's a special system token and it should be persistent (because other systems could rely on it).

**Comment by amercader** [comment]
I think being able to authenticate the calls from the demo page is a separate concern than providing tokens to partners and we probably don't want to use the same token for more than one purpose. Otherwise if someone is misusing it we need to block everyone. Also we don't want the same token to work eg on staging / production.
Yes, the token will still be public in the demo page, but it won't be hard coded in the source code. It just feels wrong that the same token will be used forever.


Also, totally unrelated but  [`token_required`](https://github.com/frictionlessdata/goodtables.io/blob/master/goodtablesio/utils/backend.py#L38) should be implemented as flask-login's [`request_loader`](https://github.com/frictionlessdata/goodtables.io/blob/master/goodtablesio/auth.py#L40)

**Comment by roll** [comment]
@amercader 
So what you propose?

**Comment by amercader** [comment]
Not having the token hard coded :)

Even if we use as you suggest there will be other places where we will need to use (eg to rate limit, add CSRF protection or whatever) so we don't want to copy the same value over and over. Read it from an env var as we do for other vars. 

And I understand that it will still be created by a migration, but we can always update it on the DB if necessary.

**Comment by roll** [comment]
@amercader 
Reading from env vars with defaults to initial migration values. Good to go? Still not agree on how you consider this token (as a credential). But I don't really mind or could be wrong.

**Pull request merged**